controllers/gitrepository_controller.go

```diff
@@ -293,7 +361,26 @@ func (r *GitRepositoryReconciler) reconcile(ctx context.Context, repository sour
 	}
 
 	message := fmt.Sprintf("Fetched revision: %s", artifact.Revision)
-	return sourcev1.GitRepositoryReady(repository, artifact, url, sourcev1.GitOperationSucceedReason, message), nil
+	return sourcev1.GitRepositoryReady(repository, artifact, includeArtifcats, url, sourcev1.GitOperationSucceedReason, message), nil
+}
+
+// hasArtifactUpdated returns true if any of the revisions in the current artifacts
+// does not match any of the artifacts in the updated artifacts
+func hasArtifactUpdated(current []*sourcev1.Artifact, updated []*sourcev1.Artifact) bool {
```

Comment:
There is probably a more efficient method of implementing this.

Reply:
Also wondering if this has a zero chance of giving false negatives, as on a quick glance it looks like that would be possible if `updated` contains two versions that exist in `current` but for artifacts from different sources.

Reply:
Would that not require two sources to have colliding hashes? I will add a url check to reduce the risk if this ever happens.

Reply:
Ok so I actually missed a small thing which could have messed things up. I added tests which verify the logic, hopefully this lays any worries to rest @hiddeco.
https://github.com/fluxcd/source-controller/blob/c418c1c0eca890c0030d99cd6eff99c9658a0823/controllers/artifact_test.go